rust/src/composepost.rs

```diff
@@ -202,6 +202,47 @@ fn postprocess_presets(rootfs_dfd: &Dir) -> Result<()> {
     Ok(())
 }
 
+fn is_overlay_whiteout(meta: &cap_std::fs::Metadata) -> bool {
+    (meta.mode() & libc::S_IFMT) == libc::S_IFCHR && meta.rdev() == 0
+}
+
+/// Auto-synthesize embedded overlayfs whiteouts; for more information
+/// see https://github.com/ostreedev/ostree/pull/2722/commits/0085494e350c72599fc5c0e00422885d80b3c660
+#[context("Postprocessing embedded overlayfs")]
+fn postprocess_embedded_ovl_whiteouts(root: &Dir) -> Result<()> {
+    const OSTREE_WHITEOUT_PREFIX: &str = ".ostree-wh.";
+    fn recurse(root: &Dir, path: &Utf8Path) -> Result<u32> {
```

Comment:
It would be good to bubble up the current-path in case of errors.
That would be a nice UX improvement, but no need to block the PR on that.

Reply:
Yeah...I think at some point we need to factor out a "walkdir" helper that does stuff like that automatically.